src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -231,7 +231,7 @@ FixedwingPositionControl::vehicle_command_poll()
 		if (vehicle_command.command == vehicle_command_s::VEHICLE_CMD_DO_GO_AROUND) {
 			// only abort landing before point of no return (horizontal and vertical)
 			if (_control_mode.flag_control_auto_enabled &&
-			    _pos_sp_triplet.current.valid &&
+			    checkPositionSetpointValid(_pos_sp_triplet.current) &&
```

Comment:
Can we check all three here (actually even better encapsulated in something like a `position_setpoint_triplet_poll()` method:
https://github.com/PX4/PX4-Autopilot/blob/a994f76636876060fc23518db29cf5e162a4cb6e/src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp#L2183-L2186

And then if lat/lon is `nan`, set the `valid` field to `false`?

At least then the interface is checked (and ideally no other method in the class should mess with this input), and we have a consistent meaning of `valid`, and not ambiguity with lat/lon.

Then the position setpoint could just be checked for validity thought the module.

Reply:
So you mean also not listen to `position_setpoint.valid `(as it comes in from Navigator) but instead set it in the FW Position controller? Then I would find it cleaner to introduce new module variables (_position_setpoint_current_valid, _previous, _next). But adding new states is seldom a good thing..
The other thing I find suboptimal is that we then bind the .valid flag strictly to the lat/lon/alt (all three of them), while actually position_setpoint.msg has other fields that may be set even if the 3d position isn't fully defined. Renaming .valid to .validPosition or something could help here, but we would need to check other usages in other modules.

BTW the flight task do exactly the same to check validity of a setpoint: https://github.com/PX4/PX4-Autopilot/blob/a193cbc393bbf66836d7cd4b3f9c93e4887ae012/src/modules/flight_mode_manager/tasks/Auto/FlightTaskAuto.cpp#L562 (wasn't even aware when I added checkPositionSetpointValid())

Reply:
>So you mean also not listen to position_setpoint.valid (as it comes in from Navigator) but instead set it in the FW Position controller? 

I share your reservations - but the whole point here is that navigator is apparently sending us nan positions on occasion. As for copies - the struct is already a local variable, so if we define one place to manipulate it, probably should be ok (at least if this is done sparingly). At least in this case it seems to me we are *correcting a wrongly set `valid` stamp.
https://github.com/PX4/PX4-Autopilot/blob/a994f76636876060fc23518db29cf5e162a4cb6e/src/modules/fw_pos_control_l1/FixedwingPositionControl.hpp#L219
So we are just sanitizing the local representation at the input interface.

The uorb message definition also does not define that lat/lon/alt are allowed to be nans for any purpose.. so I don't think we should leave it open for that case. Probably we'd actually want to check every single value that may be used at the interface (in general..). If interfaces to modules (classes) are sanitized, then internal methods should be free of needing to guard everywhere. (and calculations breaking things should be caught with unit tests .. but this is a whole different tangent with regard to fw_pos_ctrl .. needs to first be broken down to have unit tests that make any sense).

i guess the only case where there is not a necessity of a valid 3d position for `position_setpoint` is `SETPOINT_TYPE_VELOCITY`. Which illuminates a root issue here that this mega-struct is in general a poorly constructed interface that we need to split up and refactor - but until we can do that, maybe `isFinite(lat/lon/alt) && !SETPOINT_TYPE_VELOCITY` = valid for position setpoint types? and for `SETPOINT_TYPE_VELOCITY` we dont need the sanitization on position?

Reply:
Okay I added it to where the triplet is pulled. With new global variables though, I don't like overwriting the .valid flag (which is something that is set by the Navigator, and I think we should remove going forward).
d656ff83a6e5766d5477bad61dcd526b1924f9bb